hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/AvroKafkaSource.java

```diff
@@ -45,11 +46,14 @@
 
   private final KafkaOffsetGen offsetGen;
 
+  private final String useCustomDeserializerProp = "hoodie.deltastreamer.kafka.custom.avro.deserializer";
+
   public AvroKafkaSource(TypedProperties props, JavaSparkContext sparkContext, SparkSession sparkSession,
       SchemaProvider schemaProvider) {
     super(props, sparkContext, sparkSession, schemaProvider);
+    boolean useCustomDeserializer = props.getBoolean(useCustomDeserializerProp, false);
     props.put("key.deserializer", StringDeserializer.class);
-    props.put("value.deserializer", KafkaAvroDeserializer.class);
+    props.put("value.deserializer", useCustomDeserializer ? HoodieAvroKafkaDeserializer.class : KafkaAvroDeserializer.class);
```

Comment:
is it fair to assume confluent is default for avro?

Reply:
Yeah, I mean historically we have used confluent schema-registry integration with hudi. :)